src/core/lib/support/sync_posix.c

```diff
@@ -69,7 +69,12 @@ int gpr_mu_trylock(gpr_mu* mu) {
 
 /*----------------------------------------*/
 
-void gpr_cv_init(gpr_cv* cv) { GPR_ASSERT(pthread_cond_init(cv, NULL) == 0); }
+void gpr_cv_init(gpr_cv* cv) {
+  pthread_condattr_t attributes;
+  GPR_ASSERT(pthread_condattr_init(&attributes) == 0);
+  GPR_ASSERT(pthread_condattr_setclock(&attributes, CLOCK_MONOTONIC) == 0);
```

Comment:
This isn't going to work on mac... suggest bifurcating this file into sync_posix.c, sync_linux.c and making the monotonic changes only for sync_linux.c.